core/include/kernel/interrupt.h

```diff
@@ -32,17 +32,24 @@ enum itr_return {
 	ITRR_HANDLED,
 };
 
+struct itr_handler;
+
+typedef enum itr_return (*itr_handler_t)(struct itr_handler *h);
+
 struct itr_handler {
 	size_t it;
 	uint32_t flags;
-	enum itr_return (*handler)(struct itr_handler *h);
+	itr_handler_t handler;
 	void *data;
 	SLIST_ENTRY(itr_handler) link;
 };
 
 void itr_init(struct itr_chip *data);
 void itr_handle(size_t it);
 
+struct itr_handler *itr_alloc_add(size_t it, itr_handler_t handler,
+				  uint32_t flags, void *data);
+void itr_free(struct itr_handler *hdl);
```

Comment:
`itr_free()` is used nowhere in this patch series so I'd rather add it when needed.

Reply:
Like the interrupt framework provide the alloc Interface, I've prefer also provide the free Interface to the client.
This both functions integrated also  the list management Insert/Remove .

Reply:
OK, no strong opinion on this, you may leave it if you wish but I have not changed my mind ;-)

Reply:
no problem I understand the 2 POV:
-wait the need
-provide a symmetric interface alloc/free
thanks for your open mind ;-)